site-src/v1alpha2/references/cross-namespace-references.md

```diff
@@ -0,0 +1,273 @@
+ReferencePolicy is not honored by their implementations and detail which
+alternative safeguards are available. Note that this is unlikely to apply to
+ingress implementations of the API and will not apply to all mesh
+implementations.
+
+For an example of the risks involved in cross-namespace references, refer to
+[CVE-2021-25740](https://github.com/kubernetes/kubernetes/issues/103675).
+Implementations of this API need to be very careful to avoid confused deputy
+attacks. ReferencePolicy provides a safeguard for that. Exceptions MUST only
+be made by implementations that are absolutely certain that other equally
+effective safeguards are in place.
+
+## ForwardTo
+
+To enable cross-namespace forwarding, we have created a new `ObjectReference` struct that can be used in places
+where cross-namespace references are possible, and have updated the HTTPRoute ForwardTo field `BackendRef` to use the new type.
+
+```go
+
+// Fron object_reference_types.go
+
+// ObjectReference identifies an API object including its namespace.
+type ObjectReference struct {
+	// Group is the group of the referent.
+	//
+	// +kubebuilder:validation:MaxLength=253
+	Group string `json:"group"`
+
+	// Kind is kind of the referent.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Kind string `json:"kind"`
+
+	// Name is the name of the referent.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name"`
+
+	// Namespace is the namespace of the backend. When unspecified, the local
+	// namespace is inferred.
+	//
+	// Support: Core
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	// +optional
+	Namespace *string `json:"namespace,omitempty"`
+}
+
+
+// From httproute_types.go
+
+	// BackendRef is a reference to a backend to forward matched requests to. If
+	// both BackendRef and ServiceName are specified, ServiceName will be given
+	// precedence.
+	//
+	// If the referent cannot be found, the route must be dropped
+	// from the Gateway. The controller should raise the "ResolvedRefs"
+	// condition on the Gateway with the "DegradedRoutes" reason.
+	// The gateway status for this route should be updated with a
+	// condition that describes the error more specifically.
+	//
+	// If there is a cross-namespace reference to an *existing* object
+	// with no ReferencePolicy, the controller must ensure the "ResolvedRefs"
+	// condition on the Gateway is set to `status: true`, with the "RefNotPermitted"
+	// reason.
+	//
+	// Support: Custom
+	//
+	// +optional
+	BackendRef *ObjectReference `json:"backendRef,omitempty"`
+
+    // The BackendRef in the more general RouteForwardTo object in shared_types.go
+    // has also been updated.
+```
+### Conformance Level
+
+ReferencePolicy support is a "CORE" coformance level for the following
+objects:
+- HTTPRoute
+- TLSRoute
+- TCPRoute
+- UDPRoute
```

Comment:
I find this conformance guidance is overly broad and vague.
We should be more explicit here: XRoute -> cross-namespace service/secrets require RP resource.

Reply:
I've clarified a bit, does this read better?